CMakeLists.txt

```diff
@@ -252,32 +256,51 @@ if(APPLE OR CMAKE_SYSTEM_NAME MATCHES "DragonFly|FreeBSD|Linux|NetBSD|OpenBSD")
   list(APPEND uv_test_libraries util)
 endif()
 
-add_library(uv SHARED ${uv_sources})
-target_compile_definitions(uv
+if(CMAKE_SYSTEM_NAME MATCHES "CYGWIN|MSYS")
+  list(APPEND uv_defines _GNU_SOURCE)
+  list(APPEND uv_sources
+       src/unix/cygwin.c
+       src/unix/bsd-ifaddrs.c
+       src/unix/no-fsevents.c
+       src/unix/no-proctitle.c
+       src/unix/posix-hrtime.c
+       src/unix/posix-poll.c
+       src/unix/procfs-exepath.c
+       src/unix/sysinfo-loadavg.c
+       src/unix/sysinfo-memory.c)
+endif()
+
+add_library(uv_shared SHARED ${uv_sources})
+target_compile_definitions(uv_shared
   INTERFACE
     USING_UV_SHARED=1
   PRIVATE
     BUILDING_UV_SHARED=1
     ${uv_defines})
-target_compile_options(uv PRIVATE ${uv_cflags})
-target_include_directories(uv
+target_compile_options(uv_shared PRIVATE ${uv_cflags})
```

Comment:
Please don't change the names of the targets just yet. This affects anyone currently using libuv via `add_subdirectory` (I currently do this in a few small projects I'm working on and while I can fix my build, we can't fix everyone's).

I'm willing to have a discussion at least, as there are some changes in this aspect I would like to tackle at some point.